tensorflow/python/compiler/tensorrt/trt_convert.py

```diff
@@ -1307,6 +1347,88 @@ def _reset_allow_build_at_runtime(node):
     signatures[self._input_saved_model_signature_key] = self._converted_func
     save.save(self._saved_model, output_saved_model_dir, signatures)
 
+  def summary(self, line_length=160, detailed=True, print_fn=None):
+    """This method describes the results of the conversion by TF-TRT. It
+    includes information such as the name of the engine, the number of nodes per
+    engine, the input and output dtype, along with the input shape of each
+    TRTEngineOp.
+
+    Args:
+      line_length: Default line length when printing on the console. Minimum 160
+        characters long.
+      detailed: Whether or not to show the nodes inside each TRTEngineOp.
+      print_fn: Print function to use. Defaults to `print`. It will be called on
+        each line of the summary. You can set it to a custom function in order
+        to capture the string summary.
+    """
+    if not self._converted:
+      raise RuntimeError(
+        f"Impossible to call `{self.__class__.__name__}.summary()` before " \
+        f"calling {self.__class__.__name__}.convert()`."
+      )
+
+    if line_length < 160:
+      raise ValueError(f"Invalid `line_length` value has been received: " \
+                       f"{line_length}. Minimum: 160.")
+
+    if print_fn is None:
+      print_fn = print
+
+    # positions are percentage of `line_length`. positions[i]+1 is the starting
+    # position for (i+1)th field. We also make sure that the last char printed
+    # for each field is a space.
+    positions = [.22, .30, .40, .50, 1.]
+    positions = [int(line_length * p) for p in positions]
+
+    headers = [
+      "TRTEngineOP Name", "# Nodes", "Input DType", "Output Dtype",
+      "Input Shape"
+    ]
+    print_row(headers, positions, print_fn=print_fn)
+    print_fn("=" * line_length)
+
+    n_engines = 0
+    n_ops_converted = 0
+    n_ops_not_converted = 0
+
+    _graphdef = self._converted_func.graph.as_graph_def(add_shapes=True)
```

Comment:
s/_graphdef/graphdef/
It is a local variable inside a function, no need to prefix with _

Reply:
fixed